nc_time_axis/__init__.py

```diff
@@ -138,76 +151,117 @@ def __call__(self):
         return self.tick_values(vmin, vmax)
 
     def tick_values(self, vmin, vmax):
-        vmin, vmax = mtransforms.nonsingular(vmin, vmax, expander=1e-7,
-                                             tiny=1e-13)
+        vmin, vmax = mtransforms.nonsingular(
+            vmin, vmax, expander=1e-7, tiny=1e-13
+        )
 
         self.ndays = float(abs(vmax - vmin))
 
-        utime = cftime.utime(self.date_unit, self.calendar)
-        lower = utime.num2date(vmin)
-        upper = utime.num2date(vmax)
+        lower = cftime.num2date(vmin, self.date_unit, calendar=self.calendar)
+        upper = cftime.num2date(vmax, self.date_unit, calendar=self.calendar)
 
         resolution, n = self.compute_resolution(vmin, vmax, lower, upper)
 
-        if resolution == 'YEARLY':
+        def has_year_zero(year):
+            result = dict()
+            if self.calendar in self.real_world_calendars and not bool(year):
+                result = dict(has_year_zero=True)
+            return result
+
+        if resolution == "YEARLY":
             # TODO START AT THE BEGINNING OF A DECADE/CENTURY/MILLENIUM as
             # appropriate.
+
             years = self._max_n_locator.tick_values(lower.year, upper.year)
-            ticks = [cftime.datetime(int(year), 1, 1) for year in years]
-        elif resolution == 'MONTHLY':
+            ticks = [
+                cftime.datetime(
+                    int(year),
+                    1,
+                    1,
+                    calendar=self.calendar,
+                    **has_year_zero(year),
```

Comment:
Sorry not to have caught this before merge. `has_year_zero` was only introduced as a kwarg in https://github.com/Unidata/cftime/pull/234, which means the cftime requirement should be pinned: `cftime>=1.5`.

(I _think_ this ought to fix the nc-time-axis failures on https://github.com/SciTools/iris/pull/4194, although it's hard to tell if it's the root cause).